SocketBraitenburg.py

```diff
@@ -23,6 +23,23 @@ def run(self):
         while True:
             sensorValues(self.name)
 
+# ============================================
+# A thread class from https://www.tutorialspoint.com/python/python_multithreading.htm
+# This thread class represents a background thread on the robot to check if the program has quit or not
+class ExitBackgroundThread(threading.Thread):
+    def __init__(self, threadID, name, counter):
+        threading.Thread.__init__(self)
+        self.threadID = threadID
+        self.name = name
+        self.counter = counter
+
+    def run(self):
+        while True:
+            k = mySocket.recv(2048).decode()
+            if k == 'q':
+                nonlocal exitFlag
+                exitFlag = 0
+
 # =============================================
```

Comment:
ooo this is Python! This is a funny name for a commit. Why would one commit when 'debugging'?